### ISSUE TITLE
about the yago result

### ISSUE THREAD

**Issue by laixiaolian** [dair-iitd/tkbi#6]
Hello, on the yago dataset, I cannot reproduce the experimental results in the paper, especially on the time prediction task. Is the parameter not the optimal parameter?

**Comment by sushant21** [comment]
Hi, the parameters are optimal. Can you share the commands you ran, and the final model performance? 
Running the commands in the README, first to train the base model, & then to augment it using pair features, should give you the numbers reported.

**Comment by laixiaolian** [comment]
For yago dataset：
complex_base:
training: 
python main.py -d YAGO11k -m TimePlex_base -a '{"embedding_dim":200, "srt_wt":5.0, "ort_wt":5.0, "sot_wt":0.0, "emb_reg_wt":0.03}' -l crossentropy_loss_AllNeg -r 0.1 -b 1500 -x 300 -n 0 -v 1 -q 0 -y 100 -g_reg 2 -g 1.0 --perturb_time 1 --mode train -e 100 --save_dir yago_timeplex_base
Evaluating: 
python main.py -d YAGO11k -m TimePlex_base --resume_from_save "./models/yago_timeplex_base/best_valid_model.pt"  --mode test --predict_time 1 -y 40
link prediction result:  mrr:  18.24    hits@1:  10.82      hits@10:  32.08
time-interval prediction result: aeIOU@1:  12.63      gIOU@1:  23.77    TAC@1:  15.04

timeplex(pari):
training:​ 
python main.py -d YAGO11k -m TimePlex -a '{"embedding_dim":200, "model_path":"./models/yago_timeplex_base/best_valid_model.pt", "pairs_wt":3.0, "pairs_args":{"reg_wt":0.002}}' -l crossentropy_loss -r 0.05 -b 100 -x 300 -n 100 -v 1 -q 0 -y 40  -g 1.0 -bt 0 --patience 1 -e 2 --save_dir yago_timeplex
evaluating: 
python main.py -d YAGO11k -m TimePlex --resume_from_save "./models/yago_timeplex/best_valid_model.pt"  --mode test --predict_time 1 -y 40
link prediction result:  mrr:18.77    hits@1:  11.19      hits@10:  32.91
time-interval prediction result: aeIOU@1:  19.88      gIOU@1:  32.76     TAC@1:  22.53

timeplex(recurrent):
training:
python  main.py -d YAGO11k -m TimePlex -a '{"embedding_dim":200, "model_path":"./models/yago_timeplex_base/best_valid_model.pt", "recurrent_wt":5.0}' -l crossentropy_loss -r 1.0 -b 100 -x 600 -n 100 -v 1 -q 0 -y 40 -g_reg 2 -g 0.0 -bt 0 --patience 1 -e 10 --save_dir yago_timeplex
evaluating: 
python main.py -d YAGO11k -m TimePlex --resume_from_save "./models/yago_timeplex/best_valid_model.pt"  --mode test --predict_time 1 -y 40
link prediction result:  mrr:22.92    hits@1:  16.07       hits@10:  36.27
time-interval prediction result: aeIOU@1:  12.63      gIOU@1:  23.77    TAC@1:  15.04

Is it because I should use pair features and recurrent features at the same time when training timeplex? But my time-interval prediction result on timeplex_base cannot reach your paper, I dont know why. Besides, can you share the commands you ran on the icews dataset on timeplex model? On the README only the timeplex(base) commands provided.

**Comment by sushant21** [comment]
Hi, sorry for the late response-
1. Thanks for pointing out the yago issue. I looked into our old training logs, and while the training parameters are identical, the loss seems to diverge after a few epochs. Only possible reason I could come up with was a different random seed, and that seems to be the case. 
I replace the following lines in main.py ([link](https://github.com/dair-iitd/tkbi/blob/master/main.py#L23))-
    ```
    torch.manual_seed(32)
    np.random.seed(12)
    ```
    with
    ```
    torch.manual_seed(0)
    np.random.seed(0)
    ```
    and am getting the following results-
    ```
    Eval on test |M| mrr:18.27|h10:31.81|h1:10.90%
    Time prediction result: aeIOU@1: 14.64  gIOU@1: 26.27  TAC@1: 17.10
    ```
    as you can see the time prediction result has improved considerably (even better than our reported numbers!) 🤷 

2. You need to use the features together only when evaluating. For some reason training them together worsens performance, hence we train them separately and 'package' them together once trained. The script for this can be found in the repo in `scripts/` folder. ([link](https://github.com/dair-iitd/tkbi/blob/master/scripts/package_timeplex.py)). I will add this to the README.

3. As can be seen from the numbers, the gadgets bring a very marginal improvement in numbers for ICEWS datasets (we comment on the possible reasons in the paper). 
I am providing commands for the 2k dimension model (we were only able to train with the recurrent gadget)-
    ### ICEWS14-
    ```
    # Base model 
    python main.py -d icews14 -m TimePlex_base -a '{"embedding_dim":2000, "srt_wt": 5.0, "ort_wt": 5.0, "sot_wt": 5.0, "time_reg_wt":1.0, "emb_reg_wt":0.005}' -l crossentropy_loss_AllNeg -r 0.1 -b 1000 -x 2000 -n 0 -v 1 -q 0 -y 500 -g_reg 2 -g 1.0 --filter_method time-str -e 500 --flag_add_reverse 1 --save_dir icews14_timeplex_base_2k
    
    # Timeplex (with recurrent)
    python main.py -d icews14 -m TimePlex -a {"embedding_dim":2000, "model_path":"./models/icews14_timeplex_base_2k/best_valid_model.pt", "recurrent_wt":1.0} -l crossentropy_loss -r 0.1 -b 100 -x 300 -n 100 -v 1 -q 0 -y 40 -g_reg 2 -g 0.0 -bt 0 --patience 1 -e 2 --save_dir icews14_timeplex_rec_2k
    ```
    
    ### ICEWS05-15
    ```
    # Base model
    python main.py -d icews05-15 -m TimePlex_base -a '{"embedding_dim":2000, "srt_wt": 5.0, "ort_wt": 5.0, "sot_wt": 5.0, "time_reg_wt":5.0, "emb_reg_wt":0.005}' -l crossentropy_loss_AllNeg -r 0.1 -b 1000 -x 2000 -n 0 -v 1 -q 0 -y 500 -g_reg 2 -g 1.0 --filter_method time-str -e 250 --flag_add_reverse 1 --save_dir icews05-15_timeplex_base_2k
    
    # Timeplex (with recurrent)
    python main.py -d icews05-15 -m TimePlex -a '{"embedding_dim":2000, "model_path":"./models/icews05-15_timeplex_base_2k/best_valid_model.pt", "recurrent_wt":2}' -l crossentropy_loss -r 0.1 -b 60 -x 300 -n 100 -v 1 -q 0 -y 40 -g_reg 2 -g 0.0 -bt 0 --patience 1 -e 2 --save_dir icews05-15_timeplex_rec_2k
    ```

**Comment by laixiaolian** [comment]
Thanks so much! I will re-run the experiment according to your suggestion.

**Comment by laixiaolian** [comment]
Hi, I have another question. In the evaluate.py, you used ranks_o.le(11)  when calculating hits10, but I found that le() function is less than or equal to 11, which will cause the result of hits10 to be higher than actual. Is there a problem with my understanding of this function? Looking forward to your reply！

**Comment by sushant21** [comment]
This looks like a bug, yes. The hits@10 numbers will have to be re-computed for all models, including baselines :(

**Comment by laixiaolian** [comment]
Can you share the new result after you re-compute, including the result of baselines. Thanks so much!

**Comment by sushant21** [comment]
Yeah, will update once done.

**Comment by laixiaolian** [comment]
OK,  Looking forward to your reply！

**Comment by sushant21** [comment]
Hi, check README for the updated scores.

**Comment by soumen-chakrabarti** [comment]
What is the maximum / typical change in numbers?
We should also propagate the changes to the arXiv paper.

**Comment by sushant21** [comment]
Change was 1-1.5 percentage points.
@p6jain can you update on arxiv? I've updated the
latex doc.

**Comment by laixiaolian** [comment]
Hi, can you provide hits@3 indicators for TIMEPLEX and baseline in link prediction?  Looking forward to your reply！
